src/components/uui-pagination/uui-pagination.element.ts

```diff
@@ -0,0 +1,306 @@
+    // Range is number of buttons visible on either "side" of current pag button. So, if range === 5 we shall see 11 buttons in total - 5 before the curent page and 5 after. This is why we divide by 2.
+    const range = rangeBaseWidth / PAGE_BUTTON_MAX_WIDTH / 2;
+    this.range = Math.floor(range);
+  };
+
+  private _generateArrayOfNumbers(start: number, stop: number) {
+    return Array.from({ length: stop - start + 1 }, (_, i) => start + i);
+  }
+
+  private _generateVisiblePages(current: number) {
+    const start =
+      current < this._range
+        ? 1
+        : current < this.total - this._range
+        ? current - this._range
+        : this.total - this._range * 2;
+
+    const stop =
+      current <= this._range
+        ? this._range * 2 + 1
+        : current < this.total - this._range
+        ? current + this._range
+        : this.total;
+
+    const pages = this._generateArrayOfNumbers(
+      this._valueLimit(start, 1, this.total),
+      this._valueLimit(stop, 1, this.total)
+    );
+
+    return pages;
+  }
+
+  private _valueLimit(val: number, min: number, max: number) {
+    return Math.min(Math.max(val, min), max);
+  }
+
+  @queryAll('uui-button.nav-button')
+  navButtons!: Array<UUIButtonElement>;
+
+  @property()
+  label = '';
+
+  // TODO: Handle localization
+  @property({ reflect: true, attribute: 'aria-label' })
+  ariaLabel = '';
+
+  @property({ type: Number })
+  total = 1;
+
+  protected _range = 0;
+  @property({ type: Number })
+  get range() {
+    return this._range;
+  }
+
+  set range(newValue: number) {
+    const oldValue = this._range;
+    this._range = newValue <= 0 ? 1 : newValue;
+    this.visiblePages = this._generateVisiblePages(this.current);
+    this.requestUpdate('range', oldValue);
+  }
+
+  @state()
+  visiblePages: number[] = [];
+
+  protected _current = 1;
+  @property({ type: Number, reflect: true })
+  get current() {
+    return this._current;
+  }
+
+  set current(newValue: number) {
+    const oldValue = this._current;
+    this._current = this._valueLimit(newValue, 1, this.total);
+    this.visiblePages = this._generateVisiblePages(this._current);
+    this.requestUpdate('current', oldValue);
+  }
+
+  protected goToNextPage() {
+    this.current++;
+    this.dispatchEvent(new UUIPaginationEvent(UUIPaginationEvent.CHANGE));
+  }
+
+  protected goToPreviousPage() {
+    this.current--;
+    this.dispatchEvent(new UUIPaginationEvent(UUIPaginationEvent.CHANGE));
+  }
+
+  protected goToPage(page: number) {
+    this.current = page;
+    this.dispatchEvent(new UUIPaginationEvent(UUIPaginationEvent.CHANGE));
+  }
+
+  /** When having limited display of page-buttons and clicking a page-button that changes the current range, the focus stays on the position of the clicked button which is not anymore representing the number clicked, therefor we move focus to the button that represents the current page. */
```

Comment:
I added this method to enable pagination with many pages to "keep" focus on what seems to be the same button you just clicked, but isn't because the range moved. So shifting focus to the page number you just clicked.